sklearn/preprocessing/_encoders.py

```diff
@@ -692,7 +692,9 @@ def get_feature_names(self, input_features=None):
 
         feature_names = []
         for i in range(len(cats)):
-            names = [input_features[i] + "_" + str(t) for t in cats[i]]
+            if type(input_features) is not str:
+                warnings.warn("Features names aren't strings, errors could occur")
```

Comment:
Please provide always the observed invalid value in error message, for instance:

```python
    warnings.warn(
        "Unexpected type of feature name feature_names[{i}]={feature_names[i]!r}. "
        "Converting to str now. To silence this warning, please make sure encode "
        "feature names as `str` from the start."
    )
```